tools/icalconverter/main.cpp

```diff
@@ -0,0 +1,865 @@
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Nemo Mobile nor the names of its contributors
+ *     may be used to endorse or promote products derived from this
+ *     software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE."
+ */
+
+#include <QCoreApplication>
+#include <QStringList>
+#include <QString>
+#include <QFile>
+#include <QBuffer>
+#include <QDataStream>
+#include <QtDebug>
+
+#include <memorycalendar.h>
+#include <extendedcalendar.h>
+#include <extendedstorage.h>
+#include <icalformat.h>
+#include <incidence.h>
+#include <event.h>
+#include <todo.h>
+#include <journal.h>
+#include <attendee.h>
+#include <kdatetime.h>
+
+#define LOG_DEBUG(msg) if (printDebug) qDebug() << msg
+
+#define COPY_IF_NOT_EQUAL(dest, src, get, set) \
+{ \
+    if (dest->get != src->get) { \
+        dest->set(src->get); \
+    } \
+}
+
+#define RETURN_FALSE_IF_NOT_EQUAL(a, b, func, desc) {\
+    if (a->func != b->func) {\
+        LOG_DEBUG("Incidence" << desc << "" << "properties are not equal:" << a->func << b->func); \
+        return false;\
+    }\
+}
+
+#define RETURN_FALSE_IF_NOT_EQUAL_CUSTOM(failureCheck, desc, debug) {\
+    if (failureCheck) {\
+        LOG_DEBUG("Incidence" << desc << "properties are not equal:" << desc << debug); \
+        return false;\
+    }\
+}
+
+namespace {
+    mKCal::Notebook::Ptr defaultLocalCalendarNotebook(mKCal::ExtendedStorage::Ptr storage)
+    {
+        mKCal::Notebook::List notebooks = storage->notebooks();
+        Q_FOREACH (const mKCal::Notebook::Ptr nb, notebooks) {
+            if (nb->isMaster() && !nb->isShared() && nb->pluginName().isEmpty()) {
+                // assume that this is the default local calendar notebook.
+                return nb;
+            }
+        }
+        qWarning() << "No default local calendar notebook found!";
+        return mKCal::Notebook::Ptr();
+    }
+}
+
+namespace NemoCalendarImportExport {
+    namespace IncidenceHandler {
+        void normalizePersonEmail(KCalCore::Person *p)
```

Comment:
Do you mean assignment of some other value to ```p```? Function name ```normalizePersonEmail()``` implies it will do something with object pointed by ```p```.